src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/InheritDocTaglet.java

```diff
@@ -36,70 +39,80 @@
         Content replacement = writer.getOutputInstance();
         BaseConfiguration configuration = writer.configuration();
         Messages messages = configuration.getMessages();
         Utils utils = configuration.utils;
         CommentHelper ch = utils.getCommentHelper(e);
-        Taglet inheritableTaglet = holderTag == null
+        Taglet taglet = holderTag == null
                 ? null
                 : configuration.tagletManager.getTaglet(ch.getTagName(holderTag));
-        if (inheritableTaglet != null &&
-            !(inheritableTaglet instanceof InheritableTaglet)) {
-                String message = utils.getSimpleName(e) +
-                    ((utils.isExecutableElement(e))
-                        ? utils.flatSignature((ExecutableElement)e, writer.getCurrentPageElement())
-                        : "");
-                //This tag does not support inheritance.
-                var path = writer.configuration().utils.getCommentHelper(e).getDocTreePath(holderTag);
-                messages.warning(path, "doclet.inheritDocWithinInappropriateTag", message);
-                return replacement;
+        if (taglet != null && !(taglet instanceof InheritableTaglet)) {
+            //This tag does not support inheritance.
+            var path = writer.configuration().utils.getCommentHelper(e).getDocTreePath(holderTag);
+            messages.warning(path, "doclet.inheritDocWithinInappropriateTag");
+            return replacement;
         }
-        Input input = new DocFinder.Input(utils, e,
-                (InheritableTaglet) inheritableTaglet, new DocFinder.DocTreeInfo(holderTag, e),
-                isFirstSentence, true);
+        var input = new DocFinder.Input(utils, e, (InheritableTaglet) taglet,
+                new DocFinder.DocTreeInfo(holderTag, e), isFirstSentence, true);
         DocFinder.Output inheritedDoc = DocFinder.search(configuration, input);
         if (inheritedDoc.isValidInheritDocTag) {
             if (!inheritedDoc.inlineTags.isEmpty()) {
                 replacement = writer.commentTagsToOutput(inheritedDoc.holder, inheritedDoc.holderTag,
-                    inheritedDoc.inlineTags, isFirstSentence);
+                        inheritedDoc.inlineTags, isFirstSentence);
             }
-
         } else {
+            // This is to assert that we don't reach here for a class declaration.
+            // Indeed, every class except for java.lang.Object has a superclass.
+            // If we ever reach here, we would need a different warning; because
+            // the below warning is about method declarations, not class declarations.
+            // Unless @inheritDoc is used inside java.lang.Object itself,
+            // which would clearly be an error, we shouldn't reach here.
+            assert !(e instanceof TypeElement typeElement)
+                    || typeElement.getSuperclass().getKind() == TypeKind.NONE;
             String message = utils.getSimpleName(e) +
```

Comment:
`message` sounds more freeform than the value might imply. I'd be tempted to call it `signature`, and replace the `""` with `e.toString()` (just in case)

Reply:
I was tempted to use `String.valueOf(e)`, but then realized that `e` is guaranteed (although implicitly) to not be `null` by the preceding call to `Utils.isExecutableElement`.

Reply:
I was tempted to use `String.valueOf(e)`, but then realized that `e` is guaranteed (although implicitly) to not be `null` by the preceding call to `Utils.isExecutableElement`.